daemon/api.go

```diff
@@ -316,7 +319,20 @@ func getSnapInfo(c *Command, r *http.Request) Response {
 		channel = localSnap.Channel
 	}
 
-	remoteSnap, _ := remoteRepo.Snap(name, channel, nil)
+	var authenticator store.Authenticator
+	overlord := c.d.overlord
+	state := overlord.State()
+	state.Lock()
+	user, err := UserFromRequest(state, r)
+	state.Unlock()
+	if err != nil {
+		return InternalError("%v", err)
+	}
+	if user != nil {
```

Comment:
User can be nil if no authorization data is available in the request. In those cases, requests are assumed to be anonymous, and no authentication is set in the requests to the store, right?